Make RecordDataFormatter->getData() return an array with the value (p…

EreMaijala opened the pull request:
…reviously returned string) and context information. I included driver and data in the context so that it's equivalent to the rendering context of the template that returned the actual value.

demiankatz commented on module/VuFind/src/VuFind/View/Helper/Root/RecordDataFormatter.php:
```diff
@@ -105,7 +105,13 @@ public function getData(RecordDriver $driver, array $spec)
                         $field = call_user_func($current['labelFunction'], $data);
                     }
-                    $result[$field] = $text;
+                    $context = isset($options['context']) ? $options['context'] : [];
```
I don't think $options is set in this context. Am I missing something?

EreMaijala commented:
I think I've addressed your change requests. The $options thing was an obvious mistake and my shame for not verifying the results properly. I removed driver and data from the context and added a test assertion to verify that context is properly transmitted.

demiankatz commented:
Thanks; this looks good to me now. I've updated the wiki page at https://vufind.org/wiki/development:architecture:record_data_formatter to reflect this expanded use of the 'context' option as well as the new return value. If you have a chance to review that, please correct any errors I have introduced (though hopefully it's fine as is).

EreMaijala commented:
Looks good to me. Thanks!
